gpytorch/kernels/kernel.py

```diff
@@ -210,7 +225,78 @@ def __call__(self, x1, x2=None, **params):
         if x2_ is None:
             x2_ = x1_
 
-        return LazyEvaluatedKernelTensor(self, x1_, x2_, **params)
+        # Check batch_dims args
+        if settings.debug.on():
+            if batch_dims is not None:
+                if not isinstance(batch_dims, tuple) or (batch_dims != (0,) and batch_dims != (0, 2)):
+                    raise RuntimeError(
+                        "batch_dims currently accepts either None, (0,), or (0, 2). Got {}.".format(batch_dims)
+                    )
+
+        if diag:
+            res = super(Kernel, self).__call__(x1_, x2_, diag=True, batch_dims=batch_dims, **params)
+
+            # Did this Kernel eat the diag option?
+            # If it does not return a LazyEvaluatedKernelTensor, we can call diag on the output
+            if not isinstance(res, LazyEvaluatedKernelTensor):
+                if res.dim() == x1_.dim() and res.shape[-2:] == torch.Size((x1_.size(-2), x2_.size(-2))):
+                    res = res.diag()
+
+            # Now we'll make sure that the shape we're getting from diag makes sense
+            if settings.debug.on():
+                # If we used batch_dims...
+                shape = self.size(x1_, x2_)
+                if batch_dims == (0, 2):
+                    if len(shape) == 2:
+                        expected_shape = torch.Size((x1_.size(-1), shape[0]))
+                    else:
+                        expected_shape = torch.Size((shape[0] * x1_.size(-1), shape[1]))
+                    if res.shape != expected_shape:
+                        raise RuntimeError(
+                            "The kernel {} is not equipped to handle batch_dims=(0, 2) "
```

Comment:
nit: name the different args in the string (being explicit is an upfront investment but will avoid bugs down the road...)

Reply:
same applies in other places